archinstall/lib/disk/device_model.py

```diff
@@ -431,6 +352,101 @@ def from_disk(cls, disk: Disk) -> _DeviceInfo:
 		)
 
 
+class DiskLayoutType(Enum):
+	Default = 'default_layout'
+	Manual = 'manual_partitioning'
+	Pre_mount = 'pre_mounted_config'
+
+	def display_msg(self) -> str:
+		match self:
+			case DiskLayoutType.Default: return str(_('Use a best-effort default partition layout'))
+			case DiskLayoutType.Manual: return str(_('Manual Partitioning'))
+			case DiskLayoutType.Pre_mount: return str(_('Pre-mounted configuration'))
+
+
+@dataclass
+class DiskLayoutConfiguration:
+	config_type: DiskLayoutType
+	device_modifications: List[DeviceModification] = field(default_factory=list)
+	lvm_config: Optional[LvmConfiguration] = None
+
+	# used for pre-mounted config
+	relative_mountpoint: Optional[Path] = None
+
+	def __post_init__(self):
+		if self.config_type == DiskLayoutType.Pre_mount and self.relative_mountpoint is None:
+			raise ValueError('Must set a relative mountpoint when layout type is pre-mount"')
```

Comment:
This was removed in 717a2237.

This keeps happening and I now have no trust that you will not remove changes to master when you merge it in to your pull request. Was anything else not merged from master correctly?

Reply:
This is a WIP PR I have not pulled master in a while

Reply:
I know it is a WIP but I still think I should note it while I am aware of it rather than forget it. Am I mistaken that the timestamp of your last commit, which I believe is a merge of master, is more recent than the commit I linked in my last comment?

Reply:
I haven't put too much effort in merging master nicely but just pushing things sp it's backed up somewhere. I'll have a look this week on getting everything up to date and making sure nothing got missed